### ISSUE TITLE
Main thread being blocked even with useWebWorker flag setted

### ISSUE THREAD

**Issue by felipeaf** [Donaldcwl/browser-image-compression#100]
This bug occurs even in examples in example folder.. To test this, i try to compress an very large image (here an 99mb png file) without reducing width or height (this will do the job harder, so the main thread will be freezed for more time. In examples apps just set minWidthOrHeight to 9999). While compression is in progress just try type "console.log('hi!')" in console. It will not work. Actually, the entire tab freezes in 15% until end.

Firefox 87.0 (64 bits) Linux, Ubuntu 20.04
Google Chrome  v90.0.4430.72 Linux 64bits

It always happens in Firefox, but sometimes is happening with Google Chrome too, with a 99mb png, 8k resolution, maxSizeMb=1 and maxWidthOrHeight=9999, with files a little smaller it seems works. It seems there is some processing in main thread even when using workers

**Comment by Donaldcwl** [comment]
@felipeaf 
Thanks for using this library.
TLDR: The reported issue is expected behavior because OffscreenCanvas is not supported by Firefox at the moment and it is required if we want to use Canvas API in WebWorker.

This library uses Canvas API to do compression, the only way to use Canvas in WebWorker is "OffscreenCanvas". However "OffscreenCanvas" is not current supported by Firefox according to https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas#browser_compatibility. It will fallback to the main thread to use the Canvas through document.createElement('canvas').
See:
https://github.com/Donaldcwl/browser-image-compression/blob/1c5e2b10d3a1073b6ff216c6b5ea133473ef6049/lib/utils.js#L317
